apps/apps.c

```diff
@@ -760,12 +794,19 @@ EVP_PKEY *load_key(const char *file, int format, int maybe_stdin,
                            &cb_data);
 #endif
     } else {
-        BIO_printf(bio_err, "bad input format specified for key file\n");
+        print_format_error(format, OPT_FMT_PEMDER | OPT_FMT_PKCS12
+#if !defined(OPENSSL_NO_RSA) && !defined(OPENSSL_NO_DSA) && !defined (OPENSSL_NO_RC4)
+                                 | OPT_FMT_MSBLOB | FORMAT_PVK
+#endif
+#ifndef OPENSSL_NO_ENGINE
+                                 | OPT_FMT_ENGINE
+#endif
+                           );
         goto end;
     }
  end:
```

Comment:
and here?

Reply:
Same as before.

Reply:
No, I just see a goto end at the end of an else block, that goes to where the normal flow of
control would continue anyways, therefore I have to ask if that is really the intention.

Reply:
Right, the "goto end" in this case is not really needed; I just left it there for symmetry with other error cases before. Yet since this may cause confusion I took it out, also in print_cert() and load_crl().